specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2024-04-01-preview/keyvault.json

```diff
@@ -0,0 +1,2411 @@
+      "properties": {
+        "direction": {
+          "type": "string",
+          "description": "The name of the access rule."
+        },
+        "addressPrefixes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Address prefixes rule."
+        },
+        "fullyQualifiedDomainNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Address prefixes rule."
+        },
+        "subscriptions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NspAccessRulesSubscriptions"
+          },
+          "description": "The subscription access rules."
+        },
+        "networkSecurityPerimeters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NspAccessRulesNsp"
+          },
+          "description": "The network security perimeter access rules."
+        }
+      },
+      "description": "The network security perimeter resource association",
+      "type": "object"
+    },
+    "NspAccessRulesSubscriptions": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The subscription id of the rule."
+        }
+      },
+      "description": "The subscription based access rule",
+      "type": "object"
+    },
+    "NspAccessRulesNsp": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The resource id of the network security perimeter."
+        },
+        "perimeterGuid": {
+          "type": "string",
+          "description": "The id of the network security perimeter."
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the network security perimeter.",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        }
+      },
+      "description": "The network security perimeter resource based access rule",
+      "type": "object"
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+    },
+    "ResourceGroupName": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Name of the resource group that contains the key vault.",
+      "x-ms-parameter-location": "method"
+    },
+    "VaultName": {
+      "name": "vaultName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "pattern": "^[a-zA-Z0-9-]{3,24}$",
+      "description": "The name of the key vault.",
+      "x-ms-parameter-location": "method"
+    },
+    "PrivateEndpointConnectionName": {
```

Comment:
>PrivateEndpointConnectionName [](http://example.com/codeflow?start=5&length=29)

can be found and reused from common-types here

[https://github.com/Azure/azure-rest-api-specs/blob/a542d7d9867b3456a99b076b8040e35fd80c0425/specification/common-types/resource-management/v5/privatelinks.json#L181C6-L181C35](https://github.com/Azure/azure-rest-api-specs/blob/a542d7d9867b3456a99b076b8040e35fd80c0425/specification/common-types/resource-management/v5/privatelinks.json#L181C6-L181C35)